openverse-api/catalog/api/models/audio.py

```diff
@@ -0,0 +1,180 @@
+from catalog.api.models.media import (
+    AbstractAltFile,
+    AbstractMedia,
+    AbstractMediaReport,
+    AbstractDeletedMedia,
+    AbstractMatureMedia,
+    AbstractMediaList,
+)
+from catalog.api.models.mixins import (
+    IdentifierMixin,
+    MediaMixin,
+    FileMixin,
+)
+
+
+class AltAudioFile(AbstractAltFile):
+    def __init__(self, attrs):
+        self.bit_rate = attrs.get('bit_rate')
+        self.sample_rate = attrs.get('sample_rate')
+        super(AltAudioFile, self).__init__(attrs)
+
+    @property
+    def sample_rate_in_khz(self):
+        return self.sample_rate / 1e3
+
+    @property
+    def bit_rate_in_kbps(self):
+        return self.bit_rate / 1e3
+
+    def __str__(self):
+        br = self.bit_rate_in_kbps
+        sr = self.sample_rate_in_khz
+        return f'<AltAudioFile {br}kbps / {sr}kHz>'
+
+    def __repr__(self):
+        return str(self)
+
+class AudioSet(IdentifierMixin, MediaMixin, FileMixin, OpenLedgerModel):
+    """
+    This is an ordered collection of audio files, such as a podcast series or
+    an album. Not to be confused with AudioList which is a many-to-many
+    collection of audio files, like a playlist or favourites library.
+
+    The FileMixin inherited by this model refers not to audio but album art.
+    """
+
+    pass
+
+
+class Audio(AbstractMedia):
+    set = models.ForeignKey(
+        help_text='Reference to set of which this track is a part.',
+        to=AudioSet,
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+    )
+    set_position = models.IntegerField(
+        blank=True,
+        null=True,
+        help_text='Ordering of the audio in the set.'
+    )
+
+    genres = ArrayField(
+        base_field=models.CharField(
+            max_length=80,
+            blank=True,
+        ),
+        null=True,
+        db_index=True,
+        help_text='The artistic style of this audio file, eg. hip-hop (music) / tech (podcasts).',
+    )
+    category = models.CharField(
+        max_length=80,
+        blank=True,
+        null=True,
+        db_index=True,
+        help_text='The category of this audio file, eg. music, podcast, news & audiobook.',
+    )
+
+    duration = models.IntegerField(
+        blank=True,
+        null=True,
+        help_text='The time length of the audio file in milliseconds.',
+    )
+    bit_rate = models.IntegerField(
+        blank=True,
+        null=True,
+        help_text='Number in bits per second, eg. 128000.',
+    )
+    sample_rate = models.IntegerField(
+        blank=True,
+        null=True,
+        help_text='Number in hertz, eg. 44100.',
+    )
+
+    alt_files = JSONField(
+        blank=True,
+        null=True,
+        help_text='JSON describing alternative files for this audio.',
+    )
+
+    @property
+    def alternative_files(self):
+        if hasattr(self.alt_files, '__iter__'):
+            return list(map(lambda val: AltAudioFile(val), self.alt_files))
```

Comment:
Could we rewrite it using list comprehension:
`[AltAudioFile(alt_file) for alt_file in self.alt_file]` ?
It is [more Pythonic and easier to read](https://docs.quantifiedcode.com/python-anti-patterns/readability/using_map_or_filter_where_list_comprehension_is_possible.html) than map/lambda expressions.

Reply:
TIL about list comprehensions 👍

Reply:
Thanks, I love comprehensions. Resolved in 47eecf5.